operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -858,4 +862,32 @@ public static boolean isMaintenanceTimeWindowsSatisfied(Reconciliation reconcili
     public static String encodeToBase64(String encode)  {
         return Base64.getEncoder().encodeToString(encode.getBytes(StandardCharsets.US_ASCII));
     }
+
+    /**
+     * Validate container's common compute resources.
+     * 
+     * @param resources Compute resources configuration.
+     * @param type Compute resource type (cpu or memory).
+     * @param owner Compute resource owner.
+     * 
+     * @return Error message or empty (no error found).
+     */
+    public static Optional<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
+        if (resources != null && resources.getRequests() != null && resources.getLimits() != null 
+                && type != null && Arrays.asList("cpu", "memory").contains(type)) {
+            Quantity cpuRequest = resources.getRequests().get(type);
+            Quantity cpuLimit = resources.getLimits().get(type);
+            if (cpuRequest != null && cpuRequest.getNumericalAmount().compareTo(BigDecimal.ZERO) != 1) {
+                return Optional.of(String.format("Invalid %s request for %s: must be > 0", type, owner));
+            }
+            if (cpuLimit != null && cpuLimit.getNumericalAmount().compareTo(BigDecimal.ZERO) != 1) {
+                return Optional.of(String.format("Invalid %s limit for %s: must be > 0", type, owner));
+            }
+            if (cpuRequest != null && cpuLimit != null && cpuRequest.getNumericalAmount().compareTo(cpuLimit.getNumericalAmount()) > 0) {
+                return Optional.of(String.format("Invalid %s request for %s: must be <= limit", type, owner));
+            }
+            return Optional.empty();
+        }
+        return Optional.empty();
```

Comment:
Having this in `else` block would make sure that nobody will forget to add a `return` statement in future modification as it will be validated by IDE / Java. This way, if someone forgets it, it will just return the empty value wihtout any warnings.